pushnotifications/src/main/java/com/pusher/pushnotifications/reporting/FCMMessageReceiver.kt

```diff
@@ -28,9 +29,21 @@ class FCMMessageReceiver : WakefulBroadcastReceiver() {
         val pusherData = gson.fromJson(pusherDataJson, PusherMetadata::class.java)
         log.i("Got a valid pusher message.")
 
+        if (context == null) {
+          log.e("Failed to get device ID (no context) - Skipping delivery tracking.")
+          return
+        }
+
+        val deviceId = DeviceStateStore(context).getDeviceId()
+        if (deviceId == null) {
+          log.e("Failed to get device ID (device ID not stored) - Skipping delivery tracking.")
```

Comment:
I have a similar opinion here as it (a) nothing is really broken and (b) it's a valid possible state that the app receives a PN but has not yet registered to our service. E.g. some other provider is sending notifications.